src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -341,33 +374,35 @@ internal IEnumerable<TypeSymbol> InferredReturnTypes()
             }
         }
 
-        private static ImmutableArray<ParameterSymbol> DelegateParameters(NamedTypeSymbol delegateType)
```

Comment:
Just a personal preference, but I don't really like this factoring out into doing everything via the delegateInvokeMethod. It feels more clear to me if all functions operate upon delegateType. (it feels kind of underhand to use the invokeMethod as a proxy/cached-lookup of the important information on the delegateType).